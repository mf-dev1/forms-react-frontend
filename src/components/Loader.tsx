import * as React from 'react';
import { withStyles, WithStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const styles = theme => ({
  progress: {
    margin: '20px auto',
    'text-align': 'center',
  },
});

type AllProps = WithStyles<typeof styles>;

export interface State {
  completed: number;
}

class Loader extends React.Component<AllProps, State> {
  private timer: any;

  constructor(props: AllProps) {
    super(props);
    this.state = { completed: 0 };
    this.timer = null;
  }

  public componentDidMount() {
    this.timer = setInterval(this.progress, 20);
  }

  public componentWillUnmount() {
    clearInterval(this.timer);
  }

  public render() {
    const { classes } = this.props;
    return (
      <div className={classes.progress}>
        <CircularProgress variant="determinate" value={this.state.completed} />
      </div>
    );
  }

  private progress = () => {
    const { completed } = this.state;
    this.setState({ completed: completed >= 100 ? 0 : completed + 1 });
  };
}

export default withStyles(styles)(Loader);
