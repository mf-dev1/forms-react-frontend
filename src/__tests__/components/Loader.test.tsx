import * as React from 'react';
import { create } from 'react-test-renderer';
import Loader from '../../components/Loader';

describe('Loader component', () => {
  test('it matches the snapshot', () => {
    const component = create(<Loader />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
