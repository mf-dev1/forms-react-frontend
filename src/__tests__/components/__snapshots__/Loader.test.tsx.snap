// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component it matches the snapshot 1`] = `
<div
  className="Loader-progress-1"
>
  <div
    aria-valuenow={0}
    className="MuiCircularProgress-root-2 MuiCircularProgress-colorPrimary-5"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "transform": "rotate(0.000deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-7"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-8"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={
          Object {
            "strokeDasharray": "126.920",
            "strokeDashoffset": "126.920px",
          }
        }
      />
    </svg>
  </div>
</div>
`;
